File: app/src/main/java/xyz/ummo/user/data/dao/DelegatedServiceDao.java
package xyz.ummo.user.data.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Ignore;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import xyz.ummo.user.data.entity.DelegatedServiceEntity;

@Dao
public interface DelegatedServiceDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertDelegatedService(DelegatedServiceEntity delegatedServiceEntity);

    @Query("SELECT * FROM delegated_service")
    LiveData<DelegatedServiceEntity> getDelegatedService();

    @Query("SELECT * FROM delegated_service WHERE service_id = :serviceId")
    LiveData<DelegatedServiceEntity> getDelegatedServiceById(String serviceId);

    @Query("SELECT * FROM delegated_service WHERE delegated_product_id = :delegatedProductId")
    LiveData<DelegatedServiceEntity> getDelegatedServiceByProductId(String delegatedProductId);

    @Update
    void updateDelegatedService(DelegatedServiceEntity delegatedServiceEntity);

    @Query("DELETE FROM delegated_service")
    void deleteAllDelegatedServices();
}

File: app/src/main/java/xyz/ummo/user/data/dao/DelegatedServiceDao.kt
package xyz.ummo.user.data.dao

import androidx.lifecycle.LiveData
import androidx.room.*
import xyz.ummo.user.data.entity.DelegatedServiceEntity

@Dao
interface DelegatedServiceDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertDelegatedService(delegatedServiceEntity: DelegatedServiceEntity?)

    @get:Query("SELECT * FROM delegated_service")
    val delegatedService: LiveData<DelegatedServiceEntity?>?

    @Query("SELECT COUNT(delegation_id) FROM delegated_service")
    fun getDelegatedServicesCount(): Int

    @Query("SELECT * FROM delegated_service WHERE delegation_id = :delegationId")
    fun getDelegatedServiceById(delegationId: String?): LiveData<DelegatedServiceEntity?>?

    @Query("SELECT * FROM delegated_service WHERE delegated_product_id = :delegatedProductId")
    fun getDelegatedServiceByProductId(delegatedProductId: String?): LiveData<DelegatedServiceEntity?>?

    @Update
    fun updateDelegatedService(delegatedServiceEntity: DelegatedServiceEntity?)

    @Query("DELETE FROM delegated_service")
    fun deleteAllDelegatedServices()
}